tests/report_status_test.go

```diff
@@ -0,0 +1,56 @@
+package nmstate_tests
+
+import (
+	"io/ioutil"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	yaml "github.com/ghodss/yaml"
+
+	nmstate_api_v1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate.io/v1"
+
+	k8s_api_v1 "k8s.io/api/apps/v1"
+	k8s_apimachinery_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s_client_v1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+)
+
+var _ = Describe("Reporting State", func() {
+	Context("periodically", func() {
+		var (
+			dsClient         k8s_client_v1.DaemonSetInterface
+			nodeNetworkState *nmstate_api_v1.NodeNetworkState
+			nodeName         string
+
+			_ = BeforeEach(func() {
+
+				By("Creating the daemon set to monitor state")
+				manifest, err := ioutil.ReadFile("manifests/examples/state-controller-ds.yaml")
```

Comment:
There is one issue. This manifest refers to image that is publicly available on quay, while on our local cluster, we want to pull it from local registry:5000 https://github.com/nmstate/kubernetes-nmstate/blob/master/cluster/sync.sh#L10.

We should generate manifests that would be used for testing on the local cluster. Otherwise, we won't test our development branch, but rather published master.